src/main/java/uk/gov/companieshouse/confirmationstatementapi/interceptor/validation/SubmissionIdValidationInterceptor.java

```diff
@@ -0,0 +1,34 @@
+package uk.gov.companieshouse.confirmationstatementapi.interceptor.validation;
+
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+import org.springframework.web.servlet.HandlerMapping;
+import uk.gov.companieshouse.confirmationstatementapi.utils.ApiLogger;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.Map;
+
+import static uk.gov.companieshouse.confirmationstatementapi.utils.Constants.CONFIRMATION_STATEMENT_ID_KEY;
+
+@Component
+public class SubmissionIdValidationInterceptor implements HandlerInterceptor {
+
+    private static final int MAX_LENGTH = 50;
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) {
+        final Map<String, String> pathVariables = (Map<String, String>) request
+                .getAttribute(HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE);
+        final var submissionId = pathVariables.get(CONFIRMATION_STATEMENT_ID_KEY);
+
```

Comment:
I think we should also have an if block here as the first check to check if submissionId is null or empty ( null check will prevent null pointer exceptions in the next if). You can use apache StringUtils.isBlank() . if it is blank/null, log message at info level saying submission id is missing, set response to be 400 Bad request and return false;